File: bindings/ruby/examples/cpu_info.rb
require 'rbsigar'

sigar = Sigar.new

infos = sigar.cpu_info_list

num = infos.length

puts num.to_s + " total CPUs.."

infos.each do |info|
    puts "Vendor........" + info.vendor
    puts "Model........." + info.model
    puts "Current Mhz..." + info.mhz.to_s
    puts "Maximum Mhz..." + info.mhz_max.to_s
    puts "Minimum Mhz..." + info.mhz_min.to_s
    puts "Cache size...." + info.cache_size.to_s
end

File: bindings/perl/examples/cpu_info.pl
#!/usr/bin/perl

use strict;
use Sigar;

my $sigar = new Sigar;

my $infos = $sigar->cpu_info_list;

my $num = scalar @$infos;

print "$num total CPUs..\n";

for my $info (@$infos) {
    print "Vendor........" . $info->vendor . "\n";
    print "Model........." . $info->model . "\n";
    print "Mhz..........." . $info->mhz . "\n";
    print "Cache size...." . $info->cache_size . "\n";
}
